src/append_only_data.rs

```diff
@@ -547,6 +568,57 @@ impl SeqAppendOnlyData<PubPermissions> {
             },
         }
     }
+
+    pub fn shell(&self, index: u64) -> Result<Self> {
+
+        if self.entry_index() < index {
```

Comment:
I would suggest replacing this with
```suggestion
        if index > self.entry_index() {
```
to make the intent more clear (checking for overflows).